crates/wallet/src/wallet/coin_selection.rs

```diff
@@ -128,9 +128,9 @@ pub type DefaultCoinSelectionAlgorithm = BranchAndBoundCoinSelection<SingleRando
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct InsufficientFunds {
     /// Sats needed for some transaction
-    pub needed: u64,
+    pub needed: Amount,
     /// Sats available for spending
-    pub available: u64,
+    pub available: Amount,
```

Comment:
I guess it would be more clear to change the doc to say "Amount needed..." and "Amount available...". We should also update the `Display` impl for `InsufficientFunds` which still assumes the amount is satoshis.

Reply:
Fixed.

Reply:
I think the `Display` impl should be changed, because `Amount` will display itself in units of BTC.

```diff
 impl fmt::Display for InsufficientFunds {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         write!(
             f,
-            "Insufficient funds: {} sat available of {} sat needed",
+            "Insufficient funds: {} available of {} needed",
             self.available, self.needed
         )
     }
 }
```

Reply:
Thanks, fixed.